src/components/Clock.tsx

```diff
@@ -0,0 +1,64 @@
+import { Component } from 'react';
+
+type Props = {
+  clockName: string
+};
+
+function getRandomName() {
+  const value = Date.now().toString().slice(-4);
+
+  return `Clock-${value}`;
+}
+
+export class Clock extends Component<Props> {
+  state = {
+    today: new Date(),
+    clockName: this.props.clockName,
```

Comment:
you should use only one source of clockName. It already exists in the App. Use it only from the props and change the name there